modules/gojira-core/src/main/java/com/flipkart/gojira/serde/handlers/json/JsonStdSerdeHandler.java

```diff
@@ -0,0 +1,328 @@
+                    .addDeserializer(List.class, new TestListDeserializer())
+                    .addSerializer(Map.class, new TestMapSerializer())
+                    .addDeserializer(Map.class, new TestMapDeserializer()));
+
+    /**
+     * @param obj object to be serialized
+     * @param <T>
+     * @return
+     * @throws TestSerdeException
+     */
+    @Override
+    public <T> byte[] serialize(T obj) throws TestSerdeException {
+        try {
+            return MAPPER.writeValueAsBytes(obj);
+        } catch (JsonProcessingException e) {
+            LOGGER.error("error serializing data.", e);
+            throw new TestSerdeException("error serializing data.", e);
+        }
+    }
+
+    /**
+     * @param bytes serialized byte[] to be de-serialized.
+     * @param clazz class to de-serialize
+     * @param <T>
+     * @return
+     * @throws TestSerdeException
+     */
+    @Override
+    public <T> T deserialize(byte[] bytes, Class<T> clazz) throws TestSerdeException {
+        try {
+            if (List.class.isAssignableFrom(clazz)) {
+                return MAPPER.readValue(bytes, (Class<T>) List.class);
+            }
+            if (Map.class.isAssignableFrom(clazz)) {
+                return MAPPER.readValue(bytes, (Class<T>) Map.class);
+            }
+            return MAPPER.readValue(bytes, clazz);
+        } catch (IOException e) {
+            LOGGER.error("error de-serializing data. class: " + clazz.toGenericString(), e);
+            throw new TestSerdeException("error de-serializing data. class: " + clazz.toGenericString(),
+                    e);
+        }
+    }
+
+    /**
+     * @param bytes serialized byte[] to be de-serialized
+     * @param obj   object which needs to be updated with the above byte[]
+     * @param <T>
+     * @throws TestSerdeException
+     */
+    @Override
+    public <T> void deserializeToInstance(byte[] bytes, T obj) throws TestSerdeException {
+        try {
+            if (List.class.isAssignableFrom(obj.getClass())) {
+                ((List) obj).clear();
+                List tmpList = MAPPER.readValue(bytes, List.class);
+                ((List) obj).addAll(tmpList);
+            } else if (Map.class.isAssignableFrom(obj.getClass())) {
+                ((Map) obj).clear();
+                Map tmpMap = MAPPER.readValue(bytes, Map.class);
+                ((Map) obj).putAll(tmpMap);
+            } else MAPPER.readerForUpdating(obj).readValue(bytes);
+        } catch (IOException e) {
+            throw new TestSerdeException("error updating object.", e);
+        }
+
+    }
+
+
+    /**
+     * @param type The class against which a custom serializer is to be added
+     * @param ser  Custom serializer for <code>type</code>
+     * @param <T>
+     */
+    public synchronized static <T> void registerSerializer(Class<T> type, JsonSerializer<T> ser) {
+        SIMPLE_MODULE.addSerializer(type, ser);
+        MAPPER.registerModule(SIMPLE_MODULE);
```

Comment:
Is this required everytime we add serializer?

Reply:
Not required. Hence, removed.